org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/InitHandler.java

```diff
@@ -211,6 +217,8 @@ public IStatus runInWorkspace(IProgressMonitor monitor) {
 				} catch (Exception e) {
 					JavaLanguageServerPlugin.logException("Initialization failed ", e);
 					connection.sendStatus(ServiceStatus.Error, e.getMessage());
+				} finally {
+					projectsManager.registerListeners();
```

Comment:
Is there a particular reason to do this since it also happens on the BaseInitHandler constructor ?

Reply:
See https://github.com/eclipse/eclipse.jdt.ls/pull/1853/commits/ffe0984c4d8f035da16350a343ed2de072fa93f6#diff-0ef00af71cdc38cf5969fb55251c0b924ceb64d6c88727308683209ffa773ea7L57

projectsManager.registerListeners() is called to early on the BaseInitHandler  constructor. It causes unnecessary gradle build.

Reply:
Whoops. Looks like I missed your removal of it. Ok, so it's just being called later now.